contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -223,8 +226,25 @@ IECore::PathMatcher readSchemaTypeSet( const pxr::UsdPrim &prim )
 	return result;
 }
 
+template<typename SchemaType>
+IECore::PathMatcher readAPISchemaSet( const pxr::UsdPrim &prim )
+{
+	IECore::PathMatcher result;
+	for( const auto &descendant : prim.GetDescendants() )
+	{
+		if( descendant.HasAPI<SchemaType>() )
+		{
+			result.addPath( USDScene::fromUSD( descendant.GetPath() ) );
+		}
+	}
+	return result;
+}
```

Comment:
The naming of this compared to the previous function doesn't feel particularly clear ( I guess we do need both, we need to check for cameras being a schema vs lights having a schema, rather than using HasAPI for both? ).

I'm wondering about something like readDescendentsMatchSchema vs readDescendantsHaveSchema?

Reply:
Yep, we definitely need both - `IsA()` won't consider mix-in API types like UsdLuxLightAPI, and `HasAPI` has static asserts that limit it to mix-in API types. I've tried something along the lines of what you suggested in https://github.com/ImageEngine/cortex/pull/1256/commits/798e7af80f1eb1d6f73a7bdf27f12dd0b22bd248.

Reply:
How's this looking to you now @danieldresser-ie? OK to merge?